bughouse_console/src/stats_handlers_tide.rs

```diff
@@ -441,3 +458,23 @@ fn format_timestamp_date_and_time(maybe_ts: Option<OffsetDateTime>) -> Option<(S
         .ok()?;
     Some((date, time))
 }
+
+fn keep_last_point_per_date(points: &mut Vec<RawStats>) {
+    if points.is_empty() {
```

Comment:
I'm curious... Did you prefer a manual loop to something like

```
points = points.into_iter()
  .filter(|p| p.last_update.is_some())
  .group_by(|p| OffsetDateTime::date(p.last_update.unwrap()).into_iter()
  .map(|(_, group)| group.last().unwrap())
  .collect();
```

as an optimization or because you find the loop to be more readable?

Reply:
I find `group_by` more readable but didn't know it does exactly this. Somehow `group_by` in my head means associative containers. This is neat, thanks!

Changed.